numpy/core/mvgavg.py

```diff
@@ -0,0 +1,52 @@
+import numpy as np
+
+def sliding_window_view(arr, shape):
+    """
+    This function was written by fnjn on github and posted to numpy/numpy.
+    It takes in an array and a shape, and then gives back an array that is
+    essentially moving around a planchette of the described shape over the
+    array.
+    """
+    n = arr.shape
+    o = n - shape + 1 # output shape
+    strides = arr.strides
+    new_shape = np.concatenate((o, shape), axis=0)
+    new_strides = np.concatenate((strides, strides), axis=0)
+    return np.lib.stride_tricks.as_strided(arr ,new_shape, new_strides)
+
+def pascal(n):
+    """returns the nth line of a pascal's triangle"""
+    return np.array([1]) if n == 1 else np.array([0,*pascal(n-1)])+np.array([*pascal(n-1),0])
+
+def triangle(n):
```

Comment:
An argue for keeping this somewhere - it somewhat fits in with the other [window functions](https://docs.scipy.org/doc/numpy/reference/routines.window.html) - although if we decide to add to that list, I'd be inclined to create an `np.window` sub-module, rather than growing the top namespace.

Reply:
Those window functions are considered a mistake, they shouldn't have been added in the first place. Certainly we don't want to add to them. `scipy.signal` is a much better place for those.

Reply:
Sounds like we need to update that doc page to recommend the scipy versions then (and the larger library of functions alongside them)